packages/tupaia-web/src/api/mutations/useLogin.ts

```diff
@@ -28,7 +32,12 @@ export const useLogin = () => {
     {
       onSuccess: () => {
         queryClient.invalidateQueries();
-        closeModal();
+        // if the user was redirected to the login page, redirect them back to the page they were on
+        if (location.state?.backgroundLocation)
```

Comment:
```suggestion
        if (location.state?.referrer)
```

Nice - this is a good idea and we shouldn't need it much since the hashes can just be appended to urls but I imagine there will be a few uses for it. Could we rename `backgroundLocation` to `referrer` as I think it describes our scenario more accurately?